src/System Application/App/File System/src/Account/FileAccountImpl.Codeunit.al

```diff
@@ -0,0 +1,247 @@
+
+        if ConnectorLogoBase64 = '' then
+            exit;
+        if not FileConnectorLogo.Get(FileAccount.Connector) then begin
+            TempBlob.CreateOutStream(OutStream);
+            Base64Convert.FromBase64(ConnectorLogoBase64, OutStream);
+            TempBlob.CreateInStream(InStream);
+            FileConnectorLogo.Connector := FileAccount.Connector;
+            FileConnectorLogo.Logo.ImportStream(InStream, StrSubstNo(ConnectorLogoDescriptionTxt, FileAccount.Connector));
+            if FileConnectorLogo.Insert() then;
+        end;
+        FileAccount.Logo := FileConnectorLogo.Logo
+    end;
+
+    procedure IsAnyAccountRegistered(): Boolean
+    var
+        FileAccount: Record "File Account";
+    begin
+        GetAllAccounts(false, FileAccount);
+
+        exit(not FileAccount.IsEmpty());
+    end;
+
+    internal procedure IsUserFileAdmin(): Boolean
+    var
+        FileScenario: Record "File Scenario";
+    begin
+        exit(FileScenario.WritePermission());
+    end;
+
+    procedure FindAllConnectors(var FileConnector: Record "File System Connector")
+    var
+        Base64Convert: Codeunit "Base64 Convert";
+        ConnectorInterface: Interface "File System Connector";
+        FileSystemConnector: Enum "File System Connector";
+        ConnectorLogoBase64: Text;
+        OutStream: Outstream;
+    begin
+        foreach FileSystemConnector in Enum::"File System Connector".Ordinals() do begin
+            ConnectorInterface := FileSystemConnector;
+            ConnectorLogoBase64 := ConnectorInterface.GetLogoAsBase64();
+            FileConnector.Connector := FileSystemConnector;
+            FileConnector.Description := ConnectorInterface.GetDescription();
+            if ConnectorLogoBase64 <> '' then begin
+                FileConnector.Logo.CreateOutStream(OutStream);
+                Base64Convert.FromBase64(ConnectorLogoBase64, OutStream);
+            end;
+            FileConnector.Insert();
+        end;
+    end;
+
+    procedure IsValidConnector(Connector: Enum "File System Connector"): Boolean
+    begin
+        exit("File System Connector".Ordinals().Contains(Connector.AsInteger()));
+    end;
+
+    procedure MakeDefault(var FileAccount: Record "File Account")
+    var
+        FileScenario: Codeunit "File Scenario";
+    begin
+        CheckPermissions();
+
+        if IsNullGuid(FileAccount."Account Id") then
+            exit;
+
+        FileScenario.SetDefaultFileAccount(FileAccount);
+    end;
+
+    procedure BrowseAccount(var FileAccount: Record "File Account")
+    var
+        FileAccountBrowser: Page "File Account Browser";
+    begin
+        CheckPermissions();
+
+        if IsNullGuid(FileAccount."Account Id") then
+            exit;
+
+        FileAccountBrowser.SetFileAccount(FileAccount);
+        FileAccountBrowser.BrowseFileAccount('');
+        FileAccountBrowser.Run();
+    end;
+
+    internal procedure CheckPermissions()
```

Comment:
same, why internal?

Reply:
the codeunit is marked as internal so everything in it is internal. Setting the procedure itself to internal does nothing.

Reply:
Make sense. I removed it.